sdk/src/account.rs

```diff
@@ -134,7 +139,21 @@ impl<'a> KeyedAccount<'a> {
         self.is_writable
     }
 
-    pub fn new(key: &'a Pubkey, is_signer: bool, account: &'a mut Account) -> Self {
+    pub fn try_account_ref(&'a self) -> Result<Ref<Account>, InstructionError> {
+        Ok(self
+            .account
+            .try_borrow()
+            .map_err(|_| InstructionError::AccountBorrowFailed)?)
```

Comment:
Maybe replace these helpers with From instances:

```rust
impl From<BorrowError> for InstructionError {
    fn from(_: BorrowError) -> Self {
        InstructionError::AccountBorrowFailed
    }
}

impl From<BorrowMutError> for InstructionError {
    fn from(_: BorrowMutError) -> Self {
        InstructionError::AccountBorrowFailed
    }
}
```

Reply:
Sounds good, there are a lot of opportunities for nice and clean helpers

Reply:
@jackcmay Looks like you may have missed by point. If you added these instances, you wouldn't need any helpers. You can just use `try_borrow()` and `try_borrow_mut()` directly, even if the function returns an `InstructionError`. The `?` syntax implies an `into()`.

Reply:
Conflicts with `instruction_processor_utils.rs`
```rust
impl<T> From<T> for InstructionError
where
    T: ToPrimitive,
{
    fn from(error: T) -> Self {
        InstructionError::CustomError(error.to_u32().unwrap_or(0xbad_c0de))
    }
}
```

Reply:
hmm, doh

Reply:
Yeah, Be nice if generics were a catch-all and did not conflict if implementation was defined for an explicit type.  Ran into the same thing trying to optimize `short_vec`